### PR TITLE
Feat/dropdown upgrades

### PR DESCRIPTION
**Pull request by mario-subo**
TODO:
- [x] map old props to new and add deprecation warning
- [x] fix bugs with docs page
- [x] Update unit tests
- [x] remove the deprecated props directly
- [x] Automatically switch to native on mobile

### CODE REVIEW

File: lib/src/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -27,25 +28,33 @@ export interface DropdownPlaceholders {
...
+    onChange: (value: DropdownValue) => void;
     className?: string;
     clearable?: boolean;
     disabled?: boolean;
     error?: string;
-    id?: string;
     label?: string;
     list: Array<DropdownItem>;
     more?: boolean;
+    /** @deprecated use `multiple` instead */
```
Comment (kherP) [inline lib/src/Dropdown/Dropdown.tsx:44]: we have quite a number of components with breaking changes, maybe we can remove this directly?

Comment (mario-subo) [reply lib/src/Dropdown/Dropdown.tsx:44]: yeah sounds good. I've added a todo in the PR for it.

File: lib/src/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -343,59 +384,51 @@ export const Dropdown: React.FC<DropdownProps> = (props: DropdownProps) => {
...
                 ref={dropdownToggleRef}
-                className={`btn btn-secondary custom-dropdown-toggle${open ? " open" : ""}${props.more ? " more mx-right" : ""}${shouldDisable ? " disabled" : ""}`}
+                className={classnames(`btn btn-secondary custom-dropdown-toggle`, { open }, { more }, { "mx-right": more }, { disabled: shouldDisable }, props?.className)}
                 id={id}
                 aria-label={`Dropdown toggle: ${getTitleLabel()}`}
                 aria-haspopup={true}
                 aria-expanded={open}
                 tabIndex={shouldDisable ? -1 : 0}
                 onClick={shouldDisable ? null : handleClickToggle}
+                hidden={isMobile()}
```
Comment (kherP) [inline lib/src/Dropdown/Dropdown.tsx:424]: maybe we can make this not to render if it's in mobile? !isMobile() && <div ../>?

File: lib/src/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -415,16 +448,17 @@ export const Dropdown: React.FC<DropdownProps> = (props: DropdownProps) => {
                 onKeyDown={handleKeyDownMenu}
                 tabIndex={0}
                 ref={dropdownMenuRef}
-                className={`dropdown-menu custom-dropdown-menu${open ? " show" : ""}${props.more ? " dropdown-menu-right" : ""}`}
+                className={`dropdown-menu custom-dropdown-menu${open ? " show" : ""}${more ? " dropdown-menu-right" : ""}`}
+                hidden={isMobile()}
```
Comment (kherP) [inline lib/src/Dropdown/Dropdown.tsx:452]: same here

**Review by omarVodiaK — APPROVED** [review]

**Pull request merged**